#!/usr/bin/env python
PACKAGE = "tf_move_relative"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("base_frame", str_t, 0, "The moving base frame, which is the center of rotation", "turtle1")
gen.add("fixed_frame", str_t, 0, "The moving fixed frame, which is the center of rotation", "world")

gen.add("max_vel_linear", double_t, 0, "Maximum linear velocity of the robot in meters/sec", 0.3, 0, 5)
gen.add("min_vel_linear", double_t, 0, "Minimum linear velocity of the robot in meters/sec", -0.3, -5, 0)
 
gen.add("max_vel_theta", double_t, 0, "Maximum angular velocity of the robot in radians/sec",  0.3, 0, 5)
gen.add("min_vel_theta", double_t, 0, "Minimum angular velocity of the robot in radians/sec", -0.3, -5, 0)

gen.add("acceleration_linear", double_t, 0, "Deceleration limit of the robot stopping at goal for linear direction in meters/sec^2", 0.2, 0, 1)
gen.add("acceleration_theta", double_t, 0, "Deceleration limit of the robot rotation in radians/sec^2", 0.2, 0, 1)

gen.add("linear_tolerance", double_t, 0, "Allowed Euclidean distance of reference frame to goal in meters", 0.05, 0)
gen.add("angular_moving_tolerance", double_t, 0, "Allowed orientation error of the robot for in-place rotation when moving in radians", 0.2, 0)
gen.add("angular_tolerance", double_t, 0, "Allowed orientation error of reference frame to goal in radians", 0.05, 0)

gen.add("rate", double_t, 0, "Loop rate in Hz", 20.0, 0.001)
gen.add("timeout", double_t, 0, "Allowed time to wait for transforms in seconds", 0.5, 0)

exit(gen.generate(PACKAGE, "tf_move_relative", "tf_MoveRelative"))